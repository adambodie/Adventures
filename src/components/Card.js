import React, { Component } from 'react'
import '../styles/home.scss'
import NewCard from './NewCard'
import CardDescription from './CardDescription'
import ExternalLink from './ExternalLink'
import InternalLink from './InternalLink'

export default class Card extends Component {
	render() {
		const { backgroundImage, color, backgroundColor, description, mainId, date, startDate, endDate, isExternal } = this.props;
		const frontStyle = {
			backgroundImage: `url('/assets/links/adventure${mainId}.jpg')`, 
			backgroundSize: 'cover', 
			backgroundPosition: 'top'
		};
		const card = (
			<div className='flip-card'>
				<div className='flip-card-inner'>
					<div className='flip-card-front' style={frontStyle}>
						<NewCard date={date} />
					</div>
					<div className='flip-card-back' style={{backgroundColor: backgroundColor}}>
						<CardDescription 
							description={description} 
							color={color} 
							startDate={startDate} 
							endDate={endDate} 
							date={date} 
						/>
					</div>
				</div>
			</div>
		)
		return(
			<React.Fragment>
				{!isExternal ? (
					<InternalLink backgroundImage={backgroundImage} card={card} />
					) : (
					<ExternalLink backgroundImage={backgroundImage} card={card} />
					)
				}
			</React.Fragment>
		);
		
	}
}
