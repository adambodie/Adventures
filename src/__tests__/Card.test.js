import React from "react"
import renderer from "react-test-renderer"
import Card from "../components/Card"

describe("Card", () => {
	it("renders correctly", () => {
		const tree = renderer.create(
			<Card 
				mainId={"1"} 
				backgroundImage={"vegas"} 
				color={"white"} 
				backgroundColor={"blue"} 
				description={"Random description"}/>
			).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
