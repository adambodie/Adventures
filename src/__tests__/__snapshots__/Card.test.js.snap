// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<a
  href="/vegas"
>
  <div
    className="flip-card"
  >
    <div
      className="flip-card-inner"
    >
      <div
        className="flip-card-front"
        style={
          Object {
            "backgroundImage": "url('https://adventures.bodiewebdesign.com/photos/links/adventure1.jpg')",
            "backgroundPosition": "top",
            "backgroundSize": "cover",
          }
        }
      />
      <div
        className="flip-card-back"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      >
        <p
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Random description
        </p>
      </div>
    </div>
  </div>
</a>
`;
